src/main/java/io/anserini/server/SearchService.java

```diff
@@ -233,6 +238,21 @@ private IndexInitializationResult initializeIndex(String prebuiltIndex) {
     }
   }
 
+  static List<Map<String, Object>> searchSharded(ShardInfo shardGroup, String query, int hits,
+    Integer efSearch, String encoder, String queryGenerator, Function<String, SearchService> serviceProvider) {
+
+    return Arrays.stream(shardGroup.getShards())
+      .parallel()
+      .map(shard -> {
+        SearchService service = serviceProvider.apply(shard.indexName);
+        return service.search(query, hits, efSearch, encoder, queryGenerator);
+      })
```

Comment:
nit, but I think I like `})` on the previous line better so the method class align.